src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -518,7 +519,7 @@ public function mirror(string $originDir, string $targetDir, \Traversable $itera
 
         // Iterate in destination folder to remove obsolete entries
         if ($this->exists($targetDir) && isset($options['delete']) && $options['delete']) {
-            $deleteIterator = $iterator;
+            $deleteIterator = 4 < \func_num_args() ? func_get_arg(4) : $iterator;
```

Comment:
```suggestion
            $deleteIterator = (4 < \func_num_args() ? func_get_arg(4) : null) ?? $iterator;
```

I the caller explicitly passed `null`, the behavior should be the same as if the parameter was not specified at all.